rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/mute.kt

```diff
@@ -32,4 +39,14 @@ data class MuteAnnotationResolved(val muteLine: Int, val statementLine: Int)
  * The result of executing a mute annotation. Note that currently we have only annotations with two values.
  * This could change in the future.
  */
-data class MuteAnnotationExecuted(val programName: String, val expression: Expression, val value1Expression: Expression, val value2Expression: Expression, val result: Value, val value1Result: Value, val value2Result: Value)
+data class MuteAnnotationExecuted(val programName: String, val expression: Expression, val value1Expression: Expression, val value2Expression: Expression, val result: BooleanValue, val value1Result: Value, val value2Result: Value) {
+    fun succeeded(): Boolean = result.value
+
+    companion object {
+        private val DUMMYEXPRESSION = StringLiteral("")
```

Comment:
these fields seem suspicious. Perhaps we could make the parameters in `MuteAnnotationExecuted` nullable instead, or change this class to be abstract and have multiple subclasses

Reply:
You are right, but adding a nullability would add a mess of null checking syntax in the code.
So I created an abstract class